src/feedback/models/base.py

```diff
@@ -557,9 +596,24 @@ def save(self, *args, **kwargs):
         barcode_decode = Veranstaltung.decode_barcode(self.barcode)
         verst_obj = Veranstaltung.objects.get(pk=barcode_decode['veranstaltung'])
         self.veranstaltung = verst_obj
+        self.veranstaltung.log(False, True)
 
         if (barcode_decode['tutorgroup'] >= 1):
             tutorgroup = Tutor.objects.get(veranstaltung=verst_obj, nummer=barcode_decode['tutorgroup'])
             self.tutorgroup = tutorgroup
 
         super(BarcodeScannEvent, self).save(*args, **kwargs)
+
+
+class Log(models.Model):
+    veranstaltung = models.ForeignKey(Veranstaltung, null=True)
+    timestamp = models.DateTimeField(auto_now_add=True)
+    status = models.IntegerField(choices=Veranstaltung.STATUS_CHOICES, default=Veranstaltung.STATUS_ANGELEGT)
+    verursacher = models.CharField(max_length=30, blank=True)
```

Comment:
Please use two foreign Key relations for verursacher instead of a CharField. If the verursacher is a user or a Veranstalter you should link to the [Django auth user model](https://docs.djangoproject.com/en/1.10/ref/contrib/auth/#user-model). There should be a Veranstalter user in auth ;)
If the Verursacher is a Barcode scanner you should link to the Barcode scanner.